src/uris/deepLinks/deepLinkService.ts

```diff
@@ -1078,15 +1082,16 @@ export class DeepLinkService implements Disposable {
 				}
 				case DeepLinkServiceState.GoToTarget: {
 					// Need to re-fetch the remotes in case we opened in a new window
-
 					if (targetType === DeepLinkType.Repository) {
 						if (
 							this._context.action === DeepLinkActionType.Switch ||
 							this._context.action === DeepLinkActionType.SwitchToPullRequest ||
 							this._context.action === DeepLinkActionType.SwitchToPullRequestWorktree ||
 							this._context.action === DeepLinkActionType.SwitchToAndSuggestPullRequest
 						) {
-							action = DeepLinkServiceAction.OpenSwitch;
+							action = this._context.openedInNewWindow
```

Comment:
I see now what we're trying to go for with this "opened in new window". The problem is that there are other cases (like "switch to branch" with a non-pr branch) where we would open a new window as a part of the deep link flow. In those cases:

1. If this context was not set, it would make the code confusing, based on how it is named.
2. If this context was set, the deep link would not work because no PR is involved.

I have an idea to make this entire thing a bit easier on ourselves (until I move this processing out of deep links entirely in the future).

To show the PR changes, ultimately we just need the repo path (string), the base ref (string), the head ref (string), and the PR's title (string) and then we run `openComparisonChanges` on that, right?

So why don't we just encode `repoPath`, `pullRequestBaseRef` and `pullRequestHeadRef` into the query of the deep link for the PR in `getPullRequestBranchDeepLink`, and then just use those directly in the "open all PR changes" case? The entire query (`URLSearchParams`) should already be in `this._context.params`.

Then we can just extract them (null check and move to error state if they don't exist) and we wouldn't need to send PR ID, nor would we need to re-fetch the pull request object, nor the remote, etc.